proposals/0216-widget-support.md

```diff
@@ -246,73 +263,268 @@ The RPC `Show` and `SetDisplayLayout` are identified as the only screen related
+
+The details about the currently selected widget template are provided using the same display capabilities struct. It is expected that the HMI will return with fewer text fields (only main field 1 and 2), shorter text field width and smaller image resolutions. This solution reuses the existing API, to provide widget template details as mentioned.
+
+Examples for widget screens:
+
+| template name | Example layout |
+|-|-|
+| text with graphic  | ![template](../assets/proposals/NNNN-widgets/template-text-with-graphic.png) |
+| graphic with text  | ![template](../assets/proposals/NNNN-widgets/template-graphic-with-text.png) |
+| buttons with graphic | ![template](../assets/proposals/NNNN-widgets/template-tiles-with-graphic.png) |
 
 ### Policies
 
-With above modification of `OnHMIStatus` the existing policies are compatible to widgets. The policy manager should be modified and use the HMI level of the main screen to evaluate permissions for all RPCs, except screen specific RPCs. In this case the HMI level of the targeting screen should be used for evaluation.
+With above modification of `OnHMIStatus` the existing policies are compatible to widgets. The policy manager of Core should be modified and use the HMI level of the main screen to evaluate permissions for all RPCs, except screen specific RPCs. In this case the HMI level of the targeting screen should be used for evaluation.
+
+A new functional group should be added that reflects permissions of the new RPCs.
+
+```json
+"functional_groupings": {
+  …
+  "WidgetSupport" : {
+      "rpcs":{
+          "CreateScreen":{
+              "hmi_levels":["NONE","BACKGROUND","LIMITED","FULL"]
+          }, 
+          "DeleteScreen":{
+              "hmi_levels":["NONE","BACKGROUND","LIMITED","FULL"]
+          }
+      }
+  }
+}
+```
+
+This functional group should be provided to the app developers as a conditional feature, they can request. This should be done on the SmartDeviceLink developer portal when managing the app and the permissions. The web portal needs to have a checkbox additionally to the other checkboxes. The checkbox could be labeled "Allow creating widgets".
+
+### App Resumption
+
+If the app registers with a resumption ID and this ID is recognized by the HMI, all screens created by the application will resume including widgets and their content (text, images and buttons). The `OnSystemCapabilityUpdate` notification will provide information of the resumed screens. 
 
 ## Potential downsides
 
-No downsides identified.
+Moving screen metadata will cause more effort for OEMs and app consumers to implement this feature. The metadata needs to be send twice, in the responses but also in the system capability notification. However as the data is basically just a copy it is expected as an acceptable effort in favor of an improved API design.
```

Comment:
Update to:
Moving screen metadata will cause more effort for OEMs and app consumers to implement this feature. The metadata needs to be **sent** twice, in the responses but also in the system capability notification. **However, since the the data is basically a copy it is expected as an acceptable effort in favor of an improved API design.**